shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala

```diff
@@ -180,16 +180,19 @@ class UCX(executor: BlockManagerId, rapidsConf: RapidsConf) extends AutoCloseabl
           // TODO: remove this once ucx1.11 with random port selection would be released
           1024 + Random.nextInt(65535 - 1024)
         }
-        for (i <- 0 until maxRetries) {
-          val sockAddress = new InetSocketAddress(executor.host, startPort + i)
+        var attempt = 0
+        while (listener.isEmpty && attempt < maxRetries) {
+          val sockAddress = new InetSocketAddress(executor.host, startPort + attempt)
           try {
             ucpListenerParams.setSockAddr(sockAddress)
             listener = Option(worker.newListener(ucpListenerParams))
           } catch {
-            case ex: UcxException =>
-              logDebug(s"Failed to bind UcpListener on $sockAddress")
+            case _: UcxException =>
+              logDebug(s"Failed to bind UcpListener on $sockAddress. " +
+                s"Attempt ${attempt + 1} out of $maxRetries.")
               listener = None
           }
+          attempt += 1
```

Comment:
Nit: Move this to the top of the loop and we won't need to "fix it" in the log message.

Reply:
Made that change.